example/flake.nix

```diff
@@ -0,0 +1,82 @@
+{
+  description = "Example clojure project";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs";
+    cloure-nix-locker.url = "github:bevuta/clojure-nix-locker";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils, clojure-nix-locker, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        my-clojure-nix-locker = clojure-nix-locker.lib.customLocker {
+          inherit pkgs;
+          # XXX: For some reason this command doesn't work. There is some
+          #      clojure maven download that happens later in the build process.
+          # command = "${pkgs.clojure}/bin/clojure A:build -P";
+          command = "${pkgs.clojure}/bin/clojure -T:build uber";
+          lockfile = "./deps.lock.json";
+          src = ./.;
+        };
+      in rec {
+        packages.uberjar = pkgs.stdenv.mkDerivation {
+          pname = "clojure-nix-locker-example";
+          version = "0.1";
+
+          src = ./.;
+
+          nativeBuildInputs = with pkgs; [
+            makeWrapper
+            clojure
+            git
+          ];
+
+          # XXX: I would like to replace the beginning of this with ${my-clojure-nix-locker.shellEnv}, but that produces a non-writeable home directory. what should we do here?
+          #      - make shellEnv work like this
+          #      - does this actually need HOME to be writeable?
+          #      - make a separate writeableShellEnv
+          #      - leave it as is with this just not using shellEnv
```

Comment:
I think for the build itself the non-writeable home directory is fine. We can still explore the other options if it turns out that it's not.